src/plugins/mediaSource.js

```diff
@@ -2,13 +2,39 @@ import store from '../store'
 
 export default {
   install (Vue) {
+    function _mediaSource (source, returnAs = 'url', headers = null) {
+      return new Promise((resolve, reject) => {
+        if (headers === null) {
+          const headers = new Headers()
+          headers.append('Authorization', 'Bearer ' + store.getters.getToken)
+        }
+
+        fetch(source, { headers }).then(response => {
+          response.blob().then(blob => {
+            if (returnAs === 'base64') {
+              const reader = new FileReader()
+              reader.readAsDataURL(blob)
+              reader.onloadend = function () {
+                resolve(reader)
+              }
+            } else if (returnAs === 'url') {
+              const source = window.URL.createObjectURL(blob)
+              resolve(source)
+            }
+          })
+        }).catch(e => {
+          console.warn('media-source', e)
+          reject(e)
+        })
+      })
+    }
     // For use with <img> tags
     Vue.directive('image-source', {
       bind (el, binding) {
         // Set an empty image while resolving
         el.setAttribute('src', 'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAQAAAC1HAwCAAAAC0lEQVR42mNkYAAAAAYAAjCB0C8AAAAASUVORK5CYII=')
 
-        this.mediaSource(binding.value).then(dataImage => {
+        _mediaSource(binding.value).then(dataImage => {
```

Comment:
Could have been done by introducing `vnode` as param next to `binding` and then replacing `this` with `vnode.context` but I guess it doesn't really matter if it's that or this solution.

Reply:
I did not deeper look into vue plugins - the only thing I know - this.mediaSource does not work ;-)

Reply:
Sure, no worries, just mentioning it as another option how to do that :wink: